### ISSUE TITLE
_FILE environement variables don't work for Cloudflare

### ISSUE THREAD

**Issue by alrayyes** [containous/traefik#4840]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

Passed CF_API_EMAIL_FILE=/run/secrets/cf_api_email & CF_API_KEY_FILE=/run/secrets/cf_api_key in docker-compose.yml

### What did you expect to see?

ACME to properly generate SSL certificate

### What did you see instead?

time="2019-05-03T12:40:15Z" level=error msg="Unable to obtain ACME certificate for domains \"subdomain.example.com\" detected thanks to rule \"Host:subdomain.example.com\" : unable to generate a certificate for the domains [subdomain.example.com]: acme: Error -> One or more domains had a problem:\n[subdomain.example.com] [subdomain.example.com] acme: error presenting token: cloudflare: failed to find zone example.com.: ListZones command failed: error from makeRequest: HTTP request failed: Get https://api.cloudflare.com/client/v4/zones?name=example.com: net/http: invalid header field value \"xxxxxxxxxxxxxxxxxxxx\\n\" for key X-Auth-Key\n"

### Output of `traefik version`: (_What version of Traefik are you using?_)

Version:      v1.7.11
Codename:     maroilles
Go version:   go1.11.9
Built:        2019-04-26_08:42:33AM
OS/Arch:      linux/amd64

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
constraints = ["tag==external"]
insecureSkipVerify = true

defaultEntryPoints = ["http", "https"]

[entryPoints]
    [entryPoints.http]
    address = ":80"
        [entryPoints.http.redirect]
        entryPoint = "https"
    [entryPoints.https]
    address = ":443"
    [entryPoints.https.tls]



[docker]

domain = "example.com"
watch = true
exposedByDefault = false

[acme]
email = "example@example.com"
storage = "acme.json"
entryPoint = "https"
onHostRule = true
[acme.dnsChallenge]
provider = "cloudflare"
```

docker-compose.yml:
```yaml
version: "3.7"
services:
    reverse-proxy:
        image: andthensome/traefik-external
        container_name: traefik-external
        restart: always
        command: --docker --metrics --metrics.prometheus
        ports:
            - "10.10.10.10:80:80"
            - "10.10.10.10:443:443"
        volumes:
            - /var/run/docker.sock:/var/run/docker.sock
            - /media/traefik/acme-external.json:/acme.json
        environment:
            CF_API_EMAIL: example@example.com
            CF_API_KEY_FILE: /run/secrets/cf_api_key
        networks:
            - proxy
            - proxy-internal
            - prometheus-metrics
        labels:
            - traefik.tags=internal
            - traefik.frontend.rule=Host:traefik.example.com
            - traefik.port=8080
            - traefik.docker.network=proxy-internal
            - traefik.frontend.rule=PathPrefixStrip:/external
            - traefik.frontend.headers.contentSecurityPolicy=default-src 'self'; script-src 'self' 'unsafe-eval' 'unsafe-inline'; style-src 'self' 'unsafe-inline'; object-src 'none'; frame-ancestors 'none'; form-action 'none'; base-uri 'self'
            - traefik.frontend.headers.STSSeconds=315360000
            - traefik.frontend.headers.STSIncludeSubdomains=true
            - traefik.frontend.headers.customFrameOptionsValue=SAMEORIGIN
            - traefik.frontend.headers.browserXSSFilter=true
            - traefik.frontend.headers.contentTypeNosniff=true
        secrets:
            - cf_api_email
            - cf_api_key

secrets:
    cf_api_email:
        file: cf_api_email.txt
    cf_api_key:
        file: cf_api_key.txt

networks:
    proxy:
        name: proxy
    proxy-internal:
        name: proxy-internal
    prometheus-metrics:
        name: prometheus-metrics
        internal: true
```

cf_api_email.txt / cf_api_key.txt don't have any newline characters. Key is 100% alphanumeric. when passing key / email as CF_API_EMAIL / CF_API_KEY Traefik generates SSL certificate as expected.

**Comment by ldez** [comment]
Related to https://github.com/go-acme/lego/issues/871

**Comment by traefiker** [comment]
Closed by #4910.
